include/NAS2D/Common.h

```diff
@@ -45,8 +46,8 @@ std::pair<std::string, std::string> splitOnLast(const std::string& str, char del
 std::string join(std::vector<std::string> strs, char delim, bool skip_empty = true);
 std::string join(std::vector<std::string> strs, bool skip_empty = true);
 std::string trimWhitespace(std::string string);
-bool startsWith(const std::string& string, const std::string& start) noexcept;
-bool endsWith(const std::string& string, const std::string& end) noexcept;
+bool startsWith(const std::string_view& string, const std::string& start) noexcept;
+bool endsWith(const std::string_view& string, const std::string& end) noexcept;
```

Comment:
Ahh, excellent, I like this. That's true, this was added in C++17, which we're now using.

Reply:
Funny thing, when looking this up, I found that in C++20 there's going to be a `starts_with` and `ends_with` added to the standard library (that *only* operates on `string_view`s). So this is just a temporary work around until that gets implemented by at least MSVC. 

The string versions internally convert to a string_view:
https://en.cppreference.com/w/cpp/string/basic_string/starts_with
https://en.cppreference.com/w/cpp/string/basic_string/ends_with
https://en.cppreference.com/w/cpp/string/basic_string_view/starts_with
https://en.cppreference.com/w/cpp/string/basic_string_view/ends_with

Reply:
A `string_view` should encompass read only access to all of the common string types. If an algorithm is designed to work with a `string_view`, it should work for any of the string types, be they `std::string`, or null terminated `char*` arrays. It's basically a `const char*` and a `size` field tied together. Very cheap to construct or pass by-value.